dali/base/dafdesc.cpp

```diff
@@ -1340,6 +1352,13 @@ class CFileDescriptor:  public CFileDescriptorBase, implements ISuperFileDescrip
             if (_interleaved)
                 *_interleaved = interleaved;
         }
+        const char * buffer = mb.toByteArray()+mb.getPos();
+        const char * historyStr = "History";
+        bool hasEnoughData = (mb.length() - mb.getPos()) > strlen(historyStr);
```

Comment:
Would be better if 'remaining()' test proceeded const char * buffer = mb.toByteArray()+mb.getPos();